Sources/Bootstrap/Tags/ButtonGroup.swift

```diff
@@ -0,0 +1,72 @@
+import Leaf
+import TemplateKit
+
+/// Bootstrap Button Group Tag
+public final class ButtonGroupTag: TagRenderer {
+    enum GroupKeys: String {
+        case standard = "btn-group"
+        case vertical = "btn-group-vertical"
+    }
+
+    private static let paramCount: Int = 3
+
+    public func render(tag: TagContext) throws -> EventLoopFuture<TemplateData> {
+
+        let body = try tag.requireBody()
+
+        guard tag.parameters.count <= ButtonGroupTag.paramCount else {
+            throw tag.error(reason: "Wrong Parameter Count: \(tag.parameters.count) Expecting <= \(ButtonGroupTag.paramCount)")
+        }
+
+        var group = GroupKeys.standard.rawValue
+        var classes: String?
+        var aria = UUID().uuidString
+
+        if tag.parameters.count > 0 {
+            guard let param = tag.parameters[0].bool else {
+                throw tag.error(reason: "Wrong type given (expected a bool): \(type(of: tag.parameters[0]))")
+            }
+
+            if param == true {
+                group = GroupKeys.vertical.rawValue
+            }
+        }
+
+        if tag.parameters.count > 1 {
+            guard let param = tag.parameters[1].string else {
+                throw tag.error(reason: "Wrong type given (expected a string): \(type(of: tag.parameters[0]))")
+            }
+
+            if param.count > 0 {
+                classes = param
+            }
+        }
+
+        if tag.parameters.count > 2 {
+            guard let param = tag.parameters[2].string else {
+                throw tag.error(reason: "Wrong type given (expected a string): \(type(of: tag.parameters[0]))")
+            }
+
+            if param.count > 0 {
+                aria = param
+            }
+        }
+
+        return tag.serializer.serialize(ast: body).map(to: TemplateData.self) { er in
+            let body = String(data: er.data, encoding: .utf8) ?? ""
+
+            /// Button Group isn't useful without a body
+            guard body.count > 0 else {
```

Comment:
I believe there exists a `requireBody` helper function for this purpose

Reply:
I believe that just checks if you have { }  which is body with count of zero..
The ButtonGroup makes no sense with out:
 ```{
button
button}```

Reply:
Ah, you're right!

Reply:
probably another opportunity to have TemplateKit enhancement to do a more optional strict check